extensions/binding/org.eclipse.smarthome.binding.bosesoundtouch/src/main/java/org/eclipse/smarthome/binding/bosesoundtouch/internal/discovery/SoundTouchDiscoveryParticipant.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.bosesoundtouch.internal.discovery;
+
+import static org.eclipse.smarthome.binding.bosesoundtouch.BoseSoundTouchBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SoundTouchDiscoveryParticipant} is responsible processing the
+ * results of searches for mDNS services of type _soundtouch._tcp.local.
+ *
+ * @author Christian Niessner - Initial contribution
+ * @author Thomas Traunbauer - Initial contribution
+ */
+@Component(immediate = true, configurationPid = "discovery.bosesoundtouch")
+public class SoundTouchDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(SoundTouchDiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public DiscoveryResult createResult(ServiceInfo info) {
+        DiscoveryResult result = null;
+        ThingUID uid = getThingUID(info);
+        if (uid != null) {
+
+            Map<String, Object> properties = new HashMap<>(2);
+            String label = "Bose SoundTouch";
+            try {
+                label = info.getName();
+            } catch (Exception e) {
+                // ignore and use default label
+            }
+            // remove the domain from the name
+            InetAddress[] addrs = info.getInetAddresses();
+
+            // we expect only one address per device..
+            if (addrs.length > 1) {
+                logger.warn("Bose SoundTouch device {} ({}) reports multiple addresses - using the first one: {}",
+                        info.getName(), label, Arrays.toString(addrs));
+            }
+
+            properties.put(DEVICE_PARAMETER_HOST, addrs[0].getHostAddress());
+            if (getMacAddress(info) != null) {

```

Comment:
I guess an error should be raised if the MAC address is not available (or make it not required in the config.xml)

Reply:
Not necessarily - it would still be a valid discovery result, but the user would have to figure out the MAC by himself then (and enter it in the config).